pkg/common/configmap.go

```diff
@@ -202,3 +204,39 @@ func GetConfigMapAndHashWithName(r ReconcilerCommon, configMapName string, names
 	}
 	return configMap, configMapHash, nil
 }
+
+//
+// GetConfigMap - Get config map
+//
+// if the config map is not found, requeue after requeueTimeout in seconds
+func GetConfigMap(
+	r ReconcilerCommon,
+	object client.Object,
+	cond *ospdirectorv1beta1.Condition,
+	conditionDetails ospdirectorv1beta1.ConditionDetails,
+	configMapName string,
+	requeueTimeout int,
+) (*corev1.ConfigMap, ctrl.Result, error) {
+
+	configMap := &corev1.ConfigMap{}
+	err := r.GetClient().Get(context.TODO(), types.NamespacedName{Name: configMapName, Namespace: object.GetNamespace()}, configMap)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			cond.Message = fmt.Sprintf("%s config map does not exist: %v", configMapName, err)
+			cond.Reason = ospdirectorv1beta1.ConditionReason(conditionDetails.ConditionNotFoundReason)
+			cond.Type = ospdirectorv1beta1.ConditionType(conditionDetails.ConditionNotFoundType)
+
+			LogForObject(r, cond.Message, object)
+
+			return configMap, ctrl.Result{}, err
```

Comment:
From description we want to reconcile when not found, so should be:

`return configMap, ctrl.Result{RequeueAfter: time.Duration(requeueTimeout) * time.Second}, nil`

Reply:
yes, well spotted!